src/collection.js

```diff
@@ -1422,6 +1439,19 @@ export class CollectionTransaction {
     return { data: existing, permissions: {} };
   }
 
+  /**
+   * Soft Delete all records from the local database.
+   * @param  {Object} allRecords       All the records in the collection.
+   * @return {Object}
+   */
+  deleteAll(allRecords) {
+    allRecords.forEach(record => {
+      delete record.id;
+    });
+    this._queueEvent("deleteAll", { data: [] });
+    return { data: [], permissions: {} };
```

Comment:
shouldn't we return the deleted records?

Reply:
Sure we can, it will just be returning the same as `.list()` anyway :)